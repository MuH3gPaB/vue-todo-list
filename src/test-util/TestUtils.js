import {jest} from "@jest/globals";

export const mockModule = function (module, childModules) {
    let mockedModule = {actions: {}, getters: {}, state: {}, modules: childModules};
    Object.keys(module.actions || []).forEach(key => mockedModule.actions[key] = jest.fn());
    Object.keys(module.getters || []).forEach(key => mockedModule.getters[key] = jest.fn());
    Object.keys(module.state || []).forEach(key => mockedModule.state[key] = module.state[key]);
    return mockedModule;
}
