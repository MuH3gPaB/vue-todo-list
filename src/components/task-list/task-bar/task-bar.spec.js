import {beforeEach, describe, it, jest} from "@jest/globals";
import TaskBar from './task-bar';
import ToDoListModule from '@/store/to-do-list';
import TaskModule from '@/store/task';
import {createLocalVue, mount} from "vue-test-utils";
import Vuex from 'vuex';
import {mockModule} from "@/test-util/TestUtils";


jest.mock("@/store/to-do-list");

function mockStore(task) {
    return mockModule({}, {
        ToDoListModule: mockModule(ToDoListModule, {
            Task0: mockModule(TaskModule(task))
        })
    });
}

describe('TaskBar', () => {
    let localVue;
    let store;
    let task;
    let wrapper;

    beforeEach(() => {
        task = {
            id: 0,
            text: 'Test Task',
            isDone: true,
            createdDate: '2020-10-15T19:24:52.829Z',
            dueDate: '2020-10-15T19:54:52.829Z'
        };

        localVue = createLocalVue();
        localVue.use(Vuex);
        store = new Vuex.Store(mockStore(task));

        wrapper = mount(TaskBar, {
            localVue,
            store,
            propsData: {
                taskModuleId: 'Task0'
            }
        });
    });

    it('should match snapshot', () => {
        expect(wrapper.vm.$el).toMatchSnapshot();
    });
});
