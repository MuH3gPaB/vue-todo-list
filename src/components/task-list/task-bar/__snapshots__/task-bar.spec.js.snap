// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskBar should match snapshot 1`] = `
<div
  class="task-bar"
>
  <div
    class="task"
  >
    <input
      class="form-control checkbox"
      type="checkbox"
    />
     
    <p
      class="text"
    >
      
      Test Task
    
    </p>
     
    <textarea
      class="text text-input"
      style="display: none;"
    />
     
    <div
      class="time"
    >
      <p
        class="created"
      >
        10:24 PM
      </p>
       
      <p
        class="completed"
      >
        10:54 PM
      </p>
    </div>
  </div>
   
  <div
    class="delete"
    style="display: none;"
  />
</div>
`;

exports[`TaskBar shouldRender 1`] = `
<div
  class="task-bar"
>
  <div
    class="task"
  >
    <input
      class="form-control checkbox"
      type="checkbox"
    />
     
    <p
      class="text"
    >
      
      Test Task
    
    </p>
     
    <textarea
      class="text text-input"
      style="display: none;"
    />
     
    <div
      class="time"
    >
      <p
        class="created"
      >
        10:24 PM
      </p>
       
      <p
        class="completed"
      >
        10:54 PM
      </p>
    </div>
  </div>
   
  <div
    class="delete"
    style="display: none;"
  />
</div>
`;
